### PR TITLE
Allow otlp grpc and http to be enabled with TLS on monolithic

### PR DESCRIPTION
**Pull request by rubenvp8510**
Fixes https://github.com/grafana/tempo-operator/issues/976

### CODE REVIEW

File: internal/manifests/monolithic/configmap_test.go [hunk 1]
```diff
@@ -222,9 +222,9 @@ distributor:
       protocols:
         grpc:
           tls:
-            client_ca_file: /var/run/ca-receiver/service-ca.crt
-            cert_file: /var/run/tls/receiver/tls.crt
-            key_file: /var/run/tls/receiver/tls.key
+            client_ca_file: /var/run/ca-receiver/grpc/service-ca.crt
```
Comment (pavolloffay) [inline internal/manifests/monolithic/configmap_test.go:225]: Shall we add a test to use the same certs for `otlp.protocols.http` ?

File: internal/manifests/manifestutils/tls.go [hunk 2]
```diff
@@ -117,3 +125,13 @@ func findContainerIndex(pod *corev1.PodSpec, containerName string) (int, error)
 
 	return -1, fmt.Errorf("cannot find container %s", containerName)
 }
+
+func findVolumeIndex(pod *corev1.PodSpec, volumeName string) (int, error) {
```
Comment (pavolloffay) [inline internal/manifests/manifestutils/tls.go:129]: what is the point of returning error if it is always ignored? It seems like the function could be renamed to `containsVolume(pod, volumeName) bool`

Comment (rubenvp8510) [reply internal/manifests/manifestutils/tls.go]: Done

File: internal/manifests/monolithic/configmap_test.go (hunk 1 above)
Comment (rubenvp8510) [reply internal/manifests/monolithic/configmap_test.go:331]: Done

File: internal/manifests/manifestutils/tls.go [hunk 3]
```diff
@@ -27,16 +27,20 @@ func MountCAConfigMap(
...
 	})
-	pod.Volumes = append(pod.Volumes, corev1.Volume{
-		Name: caConfigMap,
-		VolumeSource: corev1.VolumeSource{
-			ConfigMap: &corev1.ConfigMapVolumeSource{
-				LocalObjectReference: corev1.LocalObjectReference{
-					Name: caConfigMap,
+
+	hasVolume := containsVolume(pod, caConfigMap)
+	if !hasVolume {
```
Comment (iblancasa) [inline internal/manifests/manifestutils/tls.go:32]: How about?
```suggestion
	if !containsVolume(pod, caConfigMap) {
```

File: internal/manifests/manifestutils/paths.go [hunk 4]
```diff
@@ -14,6 +14,16 @@ const (
...
 
+	// ReceiverGRPCTLSCADir is the path that is mounted from the configmap for TLS for receiver.
+	ReceiverGRPCTLSCADir = "/var/run/ca-receiver/grpc"
+	// ReceiverGRPCTLSCertDir returns the mount path of the receivers certificates (for ingesting traces).
+	ReceiverGRPCTLSCertDir = TLSDir + "/receiver/grpc"
+
+	// ReceiverHTTPTLSCADir is the path that is mounted from the configmap for TLS for receiver.
+	ReceiverHTTPTLSCADir = "/var/run/ca-receiver/http"
+	// ReceiverHTTPTLSCertDir returns the mount path of the receivers certificates (for ingesting traces).
+	ReceiverHTTPTLSCertDir = TLSDir + "/receiver/http"
```
Comment (iblancasa) [inline internal/manifests/manifestutils/paths.go:25]: Wht don't we use `path.Join` here?

**Review by iblancasa — APPROVED** [review]

File: internal/manifests/manifestutils/paths.go (hunk 4 above)
Comment (rubenvp8510) [reply internal/manifests/manifestutils/paths.go:25]: We can but I don't think we need it, this are constants, we don't need to deal with double slashes, empty strings etc..

File: internal/manifests/manifestutils/tls.go (hunk 3 above)
Comment (rubenvp8510) [reply internal/manifests/manifestutils/tls.go:32]: it's fine. I can do the change

Comment (rubenvp8510) [reply internal/manifests/manifestutils/tls.go]: Changed! thanks for the review!

**Review by pavolloffay — APPROVED** [review]

**Pull request merged**